samza-core/src/main/java/org/apache/samza/operators/spec/OperatorSpecs.java

```diff
@@ -203,7 +220,7 @@ public void close() {
    */
   public static <K, M, OM, JM> JoinOperatorSpec<K, M, OM, JM> createJoinOperatorSpec(
       OperatorSpec<?, M> leftInputOpSpec, OperatorSpec<?, OM> rightInputOpSpec, JoinFunction<K, M, OM, JM> joinFn,
-      Serde<K> keySerde, Serde<M> messageSerde, Serde<OM> otherMessageSerde, long ttlMs, int opId) {
+      Serde<K> keySerde, Serde<M> messageSerde, Serde<OM> otherMessageSerde, long ttlMs, String opId) {
```

Comment:
Now that the `OperatorSpecs` appears to just return new instances from the passed-in parameters, we can get rid off it in a later refactor

Reply:
+1, it's always been this way.